src/main/java/seedu/address/logic/CommandHistory.java

```diff
@@ -0,0 +1,65 @@
+package seedu.address.logic;
+
+import java.util.ArrayList;
+
+/**
+ * Represents an abstraction for a command history data structure.
+ */
+public class CommandHistory {
+
+    private final ArrayList<String> commandHistory = new ArrayList<>();
+    private int index = 0;
+
+    /**
+     * Get the previous command text.
+     *
+     * @return the previous command text.
+     */
+    public String getPrevious() {
+        assert -1 <= index && index <= commandHistory.size();
+
+        if (index < 0) {
+            return "";
+        }
+
+        index -= 1;
+
+        if (index < 0) {
+            return "";
+        }
+
+        return commandHistory.get(index);
+    }
+
+    /**
+     * Get the next command text.
+     *
+     * @return the next command text.
+     */
+    public String getNext() {
+        assert -1 <= index && index <= commandHistory.size();
+
+        if (index >= commandHistory.size()) {
+            return "";
+        }
+
+        index += 1;
+
+        if (index >= commandHistory.size()) {
+            return "";
+        }
+
+        return commandHistory.get(index);
```

Comment:
Would it be better to change this to:
```
assert 0 <= index && index <= commandHistory.size();

index = Math.min(index + 1, commandHistory.size());

if (index == commandHistory.size()) {
    return "";
}

return commandHistory.get(index);
```
instead?

Reply:
Code logic has been changed to throw an exception.